Amq_List_Broker-main/activemq.py

```diff
@@ -0,0 +1,156 @@
+            raise e
+        elif e.response['Error']['Code'] == 'InvalidParameterException':
+            # You provided an invalid value for a parameter.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+        elif e.response['Error']['Code'] == 'InvalidRequestException':
+            # You provided a parameter value that is not valid for the current state of the resource.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+        elif e.response['Error']['Code'] == 'ResourceNotFoundException':
+            # We can't find the resource that you asked for.
+            # Deal with the exception here, and/or rethrow at your discretion.
+            raise e
+    else:
+        # Decrypts secret using the associated KMS CMK.
+        # Depending on whether the secret is a string or binary, one of these fields will be populated.
+        if 'SecretString' in get_secret_value_response:
+            secret = get_secret_value_response['SecretString']
+            secret_json = json.loads(secret)
+            return secret_json[secret_name]
+        else:
+            decoded_binary_secret = base64.b64decode(get_secret_value_response['SecretBinary'])
+             
+    
+def send_email():
+    ses = boto3.client('ses')
+    # Send Notification
+    print(f'Send Notification to {send_to}')
+    ses.send_email(
+        Source="njaiswal@cogitocorp.com",
+        Destination={
+            'ToAddresses': send_to.split(","),
+            },
+        Message={
+            'Subject': {
+                'Data': CustomerNamereturn
+                },
+            'Body': {
+                'Text': {
+                    'Data': CustomerNamereturn +"has greate count please dequeue it"
+                    }
+                }
+            }
+    )
+   
+    
+
+
+def get_request(broker_name):
+    try:
+        print("i am in get request to reterieve data")
+        
+        getPassword=get_secret(mq_password)
+        print(getPassword)
+        getUsername=get_secret(mq_user_Name)
+        print(getUsername)
+        url=geturl(broker_name) 
+        print("this is url",url)
+        req = requests.get(url,auth=(getUsername,getPassword ), verify=False)
+        root = ET.fromstring(req.content)
+        print(root)
+        decoded_response = req.content.decode('utf-8')
+        treeOne = ET.fromstring(decoded_response)
+        global CustomerNamereturn
+
+        
+        for  child in treeOne.iter():
+            CustomerName=(child.attrib.get('name'))
+            MessageCount=(child.attrib.get('consumerCount'))
+        
+            if (CustomerName is not None):
+               CustomerNamereturn=CustomerName
+            if (MessageCount is not None):
+               MessageCount = int(MessageCount)
+               print(MessageCount,CustomerNamereturn)
+               if(MessageCount >= 100):
```

Comment:
MessageCount can parameter so if tomorrow they want to increase from 100 to 150 then can be controlled from parameter instead of making changes in lambda

Reply:
done